ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/StaticQueryExecutor.java

```diff
@@ -659,20 +660,47 @@ private static TableRowsEntity proxyTo(
       final ConfiguredStatement<Query> statement,
       final ServiceContext serviceContext
   ) {
-    final RestResponse<KsqlEntityList> response = serviceContext
+    final RestResponse<List<StreamedRow>> response = serviceContext
         .getKsqlClient()
-        .makeKsqlRequest(owner.location(), statement.getStatementText());
+        .makeQueryRequest(owner.location(), statement.getStatementText());
 
     if (response.isErroneous()) {
       throw new KsqlServerException("Proxy attempt failed: " + response.getErrorMessage());
     }
 
-    final KsqlEntityList entities = response.getResponse();
-    if (entities.size() != 1) {
-      throw new RuntimeException("Boom - expected 1 entity, got: " + entities.size());
+    final List<StreamedRow> streamedRows = response.getResponse();
+    if (streamedRows.isEmpty()) {
+      throw new KsqlServerException("Invalid empty response from proxy call");
     }
 
-    return (TableRowsEntity) entities.get(0);
+    // Temporary code to convert from QueryStream to TableRowsEntity
+    // Tracked by: https://github.com/confluentinc/ksql/issues/3865
+    final Header header = streamedRows.get(0).getHeader()
+        .orElseThrow(() -> new KsqlServerException("Expected header in first row"));
+
+    final ImmutableList.Builder<List<?>> rows = ImmutableList.builder();
+
+    for (final StreamedRow row : streamedRows.subList(1, streamedRows.size())) {
```

Comment:
IIRC, the query parameter to control whether header exist has not been added yet.. So `subList(1,..)` is safe for now.

Reply:
Yup.